### PR TITLE
[gardening][Measurement] Move operators into types

### PR DESCRIPTION
**Pull request by ikesyo**
https://github.com/apple/swift-corelibs-foundation/pull/1349/files?w=1

### CODE REVIEW

**Comment by ikesyo** [comment]
@swift-ci Please test

File: Foundation/Measurement.swift [hunk 1]
```diff
@@ -72,176 +72,204 @@ extension Measurement where UnitType : Dimension {
...
-/// If the `unit` of the `lhs` and `rhs` are `==`, then this returns the result of subtracting the `value` of each `Measurement`. If they are not equal, then this will convert both to the base unit of the `Dimension` and return the result as a `Measurement` of that base unit.
-/// - returns: The result of adding the two measurements.
-public func -<UnitType : Dimension>(lhs: Measurement<UnitType>, rhs: Measurement<UnitType>) -> Measurement<UnitType> {
-    if lhs.unit == rhs.unit {
-        return Measurement(value: lhs.value - rhs.value, unit: lhs.unit)
-    } else {
-        let lhsValueInTermsOfBase = lhs.unit.converter.baseUnitValue(fromValue: lhs.value)
-        let rhsValueInTermsOfBase = rhs.unit.converter.baseUnitValue(fromValue: rhs.value)
-        return Measurement(value: lhsValueInTermsOfBase - rhsValueInTermsOfBase, unit: type(of: lhs.unit).baseUnit())
+extension Measurement where UnitType : Dimension {
```
Comment (ianpartridge) [inline Foundation/Measurement.swift]: I'm not sure that having a lot of new `extension`s is ideal. Could we move this inside the existing extension on line 47?

Comment (spevans) [reply Foundation/Measurement.swift]: Every single method seems to be in its own extension although some are using a conditional conformance so this is consistent. However maybe a followup PR could merge them so there are only two extensions.

Comment (ikesyo) [reply Foundation/Measurement.swift]: > Every single method seems to be in its own extension although some are using a conditional conformance so this is consistent. However maybe a followup PR could merge them so there are only two extensions.

Yep I'm going to do that.

Comment (ianpartridge) [reply Foundation/Measurement.swift]: OK as long as this PR isn't the end then 👍

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ianpartridge** [comment]
@swift-ci please test and merge

**Comment by ikesyo** [comment]
Thanks @ianpartridge and @spevans!

**Comment by parkera** [comment]
I believe this code is also shared with the overlay.

**Pull request merged**